app/routes/queries/listings/Results.tsx

```diff
@@ -0,0 +1,194 @@
+import type {
+  FilterFn,
+  ColumnFiltersState,
+  ColumnOrderState
+} from '@tanstack/table-core'
+import {
+  createColumnHelper,
+  getCoreRowModel,
+  getFacetedMinMaxValues,
+  getFacetedRowModel,
+  getFacetedUniqueValues,
+  getFilteredRowModel,
+  getSortedRowModel
+} from '@tanstack/table-core'
+import { useEffect, useState } from 'react'
+import { flexRender, useReactTable } from '@tanstack/react-table'
+import type { Listing, ListingResponseType } from '~/requests/GetListing'
+import type { RankingInfo } from '@tanstack/match-sorter-utils'
+import { rankItem } from '@tanstack/match-sorter-utils'
+import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/solid'
+import { classNames } from '~/utils'
+
+declare module '@tanstack/table-core' {
+  interface FilterFns {
+    fuzzy: FilterFn<unknown>
+  }
+
+  interface FilterMeta {
+    itemRank: RankingInfo
+  }
+}
+
+const fuzzyFilter: FilterFn<any> = (row, columnId, value, addMeta) => {
+  const itemRank = rankItem(row.getValue(columnId), value)
+
+  addMeta({ itemRank })
+  return itemRank.passed
+}
+
+const Results = ({ data }: { data: ListingResponseType }) => {
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+  const [globalFilter, setGlobalFilter] = useState('')
+  const [columnOrder] = useState<ColumnOrderState>([])
+
+  const columnHelper = createColumnHelper<Listing>()
+  const columns = [
```

Comment:
Jeeze I need to abstract this away. Well done managing the copy pasta :X

Reply:
lol git good

Reply:
i cri err tim